### DISCUSSION TITLE
Serialize the options from ModelMultipleChoiceFilter for use with DRF

### DISCUSSION THREAD

**Discussion by alexanderjulmer** [carltongibson/django-filter#1690] [Q&A] [answered]
Heya,

first of all, thank you for your work on this package. It really helped me a lot.

In the past, I built a library like page with `django-filter` and profited immensely from the built-in form to enable the filtering.

Now I want to create a web API for that HTML view and I am looking for a possibility to serialize what would be rendered into the forms to include it into the API's response so that I can use it to populate the filter forms of my application.

I've been going through ["Integration with DRF"](https://django-filter.readthedocs.io/en/main/guide/rest_framework.html) and the other discussion to no avail. I hope I haven't missed anything.

I see how you could implement the filtering client-side only, but that would certain filter views or server-side pagination rather hard. I guess you could also run separate queries but that seems unnecessary given that the filtered view has all the data already.

**Comment by carltongibson** [comment, marked as answer]
Hi @alexanderjulmer. 

So... DRF with use django-filter via `filter_backends` to actually filter the request, and then the Browsable API will render the form in the HTML — but all of that is server-side. 

To pass the choices to the client-side you'd need to serialise them yourself...

```
choices = filterset.form.fields["mymodel"].choices
for label, value in choices: 
    # Put them in the format you want for your frontend to consume. 
    # Either a mapping, or a pair of arrays, probably. 
```

... and then you send that to the frontend to build the form UI. 

Make sense?

**Reply by alexanderjulmer** [reply]
Yes, that makes sense, thank you.
